superset/db_engine_specs/trino.py

```diff
@@ -127,14 +129,42 @@ def get_tracking_url(cls, cursor: "Cursor") -> Optional[str]:
         return None
 
     @classmethod
-    def handle_cursor(cls, cursor: "Cursor", query: Query, session: Session) -> None:
-        """Updates progress information"""
+    def handle_cursor(cls, cursor: Cursor, query: Query, session: Session) -> None:
         tracking_url = cls.get_tracking_url(cursor)
         if tracking_url:
             query.tracking_url = tracking_url
-            session.commit()
+
+        # Adds the executed query id to the extra payload so the query can be cancelled
+        query.set_extra_json_key("cancel_query", cursor.stats["queryId"])
```

Comment:
Can we call this `trino_query_id` instead? It's more descriptive of what it really is and opens the possibility of using the id for other purposes.

Reply:
Calling it `cancel_query` is the currently established convention: It's declared [here](https://github.com/apache/superset/blob/42a5375e83087c738555a05044419b812f0f4f99/superset/sql_lab.py#L72) and later used [here](https://github.com/apache/superset/blob/42a5375e83087c738555a05044419b812f0f4f99/superset/sql_lab.py#L622) and [here](https://github.com/apache/superset/blob/42a5375e83087c738555a05044419b812f0f4f99/superset/sql_lab.py#L630-L632) . The reason why we aren't placing this logic in `get_cancel_query_id` like other engines is because the id isn't available on Trino when that method is called.